backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -196,19 +203,29 @@ public void setUserIsActive(String username, Boolean active) {
     }
 
     public void createOrganization(String name, String externalId) {
+        ApplicationList apps = _client.listApplications();
+        Optional<Application> appMatch = apps.stream().filter(a->a.getId().equals(_clientId)).findFirst();
+        if (appMatch.isEmpty()) {
+            throw new IllegalGraphqlArgumentException("Cannot add organization to Okta application with unrecognized id="+_clientId);
+        }
+        Application app = appMatch.get();
```

Comment:
```suggestion
        Application app = _client.getApplication(_clientId,  null);
        if (app == null) {
            throw new IllegalGraphqlArgumentException("Cannot add organization to Okta application with unrecognized id="+_clientId);
        }
```
Am I crazy?

Separately from that question (which is I realize quite the set-up), this should get its own special type of exception, which we log and also alert everywhere, because it's not an illegal argument, it's an incorrect configuration that we won't notice until the app fails. Though I guess the users here are our own internal admins, for now?

Back on the original topic, is sure looks like Client.getApplication always returns a non-null object, which makes me wonder what happens if you ask it for an invalid ID...

Reply:
All right, this is embarrassing.
My sequence of thoughts:
1) I want to find the application by client ID
2) Ooh, look! a getApplication(id) method!
3) Oh, darn, getApplication uses application id, not client ID
4) Gosh, looks like I have to use listApplications(query) instead
5) The query param in listApplications() only matches against app name, app label, and app credential signing kid. But I only have the client ID, shoot.
6) _Prints out the results of listApplications() with each application's properties and discovers that app id == client ID_
7) _Fails to link this new discovery back to Step 3_ 
8) Well, guess I have no choice but to iterate over the results of listApplications() and then do a filter on app id.

~Fin~

Thanks for catching this. Will make the fix in the morning.